core/math/vector2.h

```diff
@@ -104,8 +104,8 @@ struct Vector2 {
 	bool operator==(const Vector2 &p_vec2) const;
 	bool operator!=(const Vector2 &p_vec2) const;
 
-	bool operator<(const Vector2 &p_vec2) const { return (x == p_vec2.x) ? (y < p_vec2.y) : (x < p_vec2.x); }
-	bool operator<=(const Vector2 &p_vec2) const { return (x == p_vec2.x) ? (y <= p_vec2.y) : (x <= p_vec2.x); }
+	bool operator<(const Vector2 &p_vec2) const { return (Math::is_equal_approx(x, p_vec2.x)) ? (y < p_vec2.y) : (x < p_vec2.x); }
+	bool operator<=(const Vector2 &p_vec2) const { return (Math::is_equal_approx(x, p_vec2.x)) ? (y <= p_vec2.y) : (x < p_vec2.x); }
```

Comment:
Why remove the `=` in the last branch?

Reply:
Ah I get why, it can't be equal.

Reply:
Yeah, if it's equal it will never get to that point. Most of the time `<=` is implemented as `! >`. I don't know if it actually makes a performance difference on any CPU, but we could theoretically safe one NOT gate here.